.github/workflows/release.yml

```diff
@@ -0,0 +1,107 @@
+name: Release
+
+permissions:
+  contents: write
+  pull-requests: write
+
+env:
+  PANDOC_VERSION: 3.1.1.0
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - release
+
+jobs:
+  release-pr:
+    # guard against running on forks
+    if: github.repository == 'TTT-2/TTT2'
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Check if PR was merged
+        uses: actions-ecosystem/action-release-label@v1
+        id: release-label
+        if: ${{ github.event.pull_request.merged == true }}
+
+      - name: Get latest tag if a release label exists on last merged PR
+        uses: actions-ecosystem/action-get-latest-tag@v1
+        id: get-latest-tag
+        if: ${{ steps.release-label.outputs.level != null }}
+
+      - name: Bump semver tag according to detected release label
+        uses: actions-ecosystem/action-bump-semver@v1
+        id: bump-semver
+        if: ${{ steps.release-label.outputs.level != null }}
+        with:
+          current_version: ${{ steps.get-latest-tag.outputs.tag }}
+          level: ${{ steps.release-label.outputs.level }}
+
+      - name: Generate app token
+        uses: actions/create-github-app-token@v1
+        id: generate-token
+        if: ${{ steps.release-label.outputs.level != null }}
+        with:
+          app-id: ${{ secrets.APP_ID }}
+          private-key: ${{ secrets.APP_PRIVATE_KEY }}
+
+      - name: Add release heading (version, tag link and date) to CHANGELOG.md
+        run: 'sed -i "/^## Unreleased$/a\\\n## [v${TTT2_VERSION}](https://github.com/TTT-2/TTT2/tree/v${TTT2_VERSION}) ($(date +%Y-%m-%d))" CHANGELOG.md'
+        if: ${{ steps.release-label.outputs.level != null }}
+        env:
+          TTT2_VERSION: ${{ steps.bump-semver.outputs.new_version }}
+
+      - name: Generate html from changelog for cl_changes.lua
+        uses: "docker://docker.io/pandoc/core:${PANDOC_VERSION}"
+        if: ${{ steps.release-label.outputs.level != null }}
+        with:
+          args: >-
+            --output=output.html
+            --read=markdown_strict
+            <(sed -e '1,/^## \[/d' -e '/^## \[/,$d' CHANGELOG.md)
```

Comment:
i guess we try to do it without html

Reply:
the html itself is not the hard part, mixing in lua and getting proper formatting is.
i would really like to have some kind of templating like jinja2 or something, but i have yet to find a useable action for it.